shell/platform/fuchsia/dart_runner/dart_test_component_controller_v2.h

```diff
@@ -0,0 +1,157 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_SHELL_PLATFORM_FUCHSIA_DART_RUNNER_DART_TEST_COMPONENT_CONTROLLER_V2_H_
+#define FLUTTER_SHELL_PLATFORM_FUCHSIA_DART_RUNNER_DART_TEST_COMPONENT_CONTROLLER_V2_H_
+
+#include <memory>
+
+#include <fml/logging.h>
+#include <fuchsia/component/runner/cpp/fidl.h>
+#include <fuchsia/sys/cpp/fidl.h>
+#include <fuchsia/test/cpp/fidl.h>
+#include <lib/async-loop/cpp/loop.h>
+#include <lib/async/cpp/wait.h>
+#include <lib/fdio/namespace.h>
+#include <lib/sys/cpp/component_context.h>
+#include <lib/sys/cpp/service_directory.h>
+#include <lib/zx/timer.h>
+
+#include <lib/fidl/cpp/binding_set.h>
+#include "lib/fidl/cpp/binding.h"
+#include "runtime/dart/utils/mapped_resource.h"
+#include "third_party/dart/runtime/include/dart_api.h"
+
+namespace dart_runner {
+
+/// Starts a Dart component written in CFv2.
```

Comment:
Could we update the documentation for this class to explain:

1. How it's different from the DartComponentControllerV2.
2. That it implements the Suite protocol.
3. Why we chose to fork it from DartComponentControllerV2.

Reply:
Done. I think I will just file a bug to combine thee in the future. I mainly separated them to just keep implementations separate and for it to be a little easier to debug. Will combine them once dart testing is stable.

Reply:
Looks like maybe we didn't push the new comment here.

Reply:
It's there now - sorry